polygeist/lib/Dialect/Polygeist/Transforms/SYCLRaiseHost.cpp

```diff
@@ -510,10 +523,10 @@ class BufferTypeTag {
     // the type of the first parameter of the demangled constructor function
     // name.
     bool isSubBuffer = false;
-    StringRef demangled{demangledName};
-    if (auto paramStart = demangled.find('('); paramStart != StringRef::npos) {
-      if (demangled.drop_front(paramStart + 1).starts_with("sycl::_V1::buffer"))
-        isSubBuffer = true;
+    auto failureOrParams = getDemangledParameterList(constructorName);
+    if (succeeded(failureOrParams)) {
+      auto paramList = static_cast<StringRef>(*failureOrParams);
```

Comment:
Should the cast be to type `DemangleResult` ? And once you have a `DemangleResult` you can call operator() to get a StringRef ?

Reply:
I c, ok